### PR TITLE
Let Gtk handle tooltips

### PR DESCRIPTION
**Pull request by ferdnyc**
Add a `TOOLTIP_COLUMN` to the model, store the text there (when appropriate), and let the tree display tooltips automatically.

This started as a fix for this:
https://github.com/gregkh/usbview/blob/c21b10713e065dc7e2cc60cadcdbe6f1dd3bf1f5/interface.c#L41-L46

(Which can be fixed. The problem is that `gtk_tree_view_get_path_at_pos()` is looking for `bin_window`-relative coordinates, which widget coordinates (i.e. `query-tooltip` coordinates) are not. But they provide a handy function `gtk_tree_view_convert_widget_to_bin_window_coords` that can turn `x` and `y` into the proper `bin_x` and `bin_y` that should be passed to `gtk_tree_view_get_path_at_pos()`.)

But while I was looking at that, I realized that it didn't make sense to be manually handling the tooltips at all.

Gtk _also_ has a function `gtk_tree_view_set_tooltip_column` which can be used to designate a column of the data model as the source of tooltip texts. It then does all the work of setting `has-tooltip` and registering a `query-tooltip` event handler for the appropriate tree cells.

Not only is this simpler, it works better. With the previous code, because `has-tooltip` was set on the entire tree, the tooltips would move in response to _downward_ pointer motion (because the pointer intersected with the tooltip itself), but they'd ignore _upward_ pointer motion because the pointer was still hovering over the same widget annotated with the same tooltip. Now, a tooltip is individually set on each row that needs one, so the tooltips track pointer movements appropriately.

(In practice, that difference may not have been as noticeable, given that a tree will typically have at most only one or two red devices. My test build for this change forced _every_ row to be red, which made the handling of movement between adjacent, tooltipped rows easy to examine.)

### CODE REVIEW

File: usbtree.c [hunk 1]
```diff
@@ -251,13 +252,16 @@ static void DisplayDevice (struct Device *parent, struct Device *device)
...
+	if (driverAttached == FALSE) {
 		color = "red";
+		tooltip = "This device has no attached driver";
+	}
 
 	gtk_tree_store_set (treeStore, &device->leaf,
 			    NAME_COLUMN, device->name,
 			    DEVICE_ADDR_COLUMN, deviceAddr,
 			    COLOR_COLUMN, color,
+				TOOLTIP_COLUMN, tooltip,
```
Comment (ferdnyc) [inline usbtree.c:264]: Oh, sure, the one spot I actually remember to indent with tabs...

```suggestion
			    TOOLTIP_COLUMN, tooltip,
```

File: usbtree.c [hunk 2]
```diff
@@ -224,6 +224,7 @@ static void DisplayDevice (struct Device *parent, struct Device *device)
 	int		interfaceNum;
 	gboolean	driverAttached = TRUE;
 	gint		deviceAddr;
+	const gchar *tooltip = NULL;
```
Comment (ferdnyc) [inline usbtree.c:227]: Urgh.
```suggestion
	const gchar	*tooltip = NULL;
```

**Comment by ferdnyc** [comment]
Now with formatting fixes squashed in, for readability.

**Comment by gregkh** [comment]
Thank you for the changes, looks great!

**Pull request merged**